src/test/ui/issues/issue-8460-const2.rs

```diff
@@ -0,0 +1,59 @@
+// compile-flags: -C overflow-checks=on -O
```

Comment:
Maybe noopt overwrites this `-O`? That would be weird. Also weird that noopt causes *fewer* diagnostics... I'm not sure what's going on. You can undo the changes unrelated to `-(-0.0)` if you want so we can get the PR merged and open an issue about not being able to test noopt properly.

Reply:
The weird thing is that it happened in `issue-8460-const.rs` not `issue-8460-const2.rs`. `issue-8460-const.rs` has no compile flags.

Reply:
Oh. That may explain it... not sure what the best thing to do here is. Maybe instead of copying the files we should use the scheme where one test is built in multiple modes? I forget what the appropriate flags are, but incremental has loads of them

Reply:
Or could we add the `-O` flag to `issue-8460-const.rs`? The other tests we touch have this flag but that hasn't.

Reply:
If that solves it, sure

Reply:
I added the flag to the test, could we run `bors try`?

Reply:
You can test this locally by setting https://github.com/rust-lang/rust/blob/b9de4ef89e0e53099a084001b26ec3207c5f8391/config.toml.example#L355 to false in your `config.toml`

Unfortunately `bors try` won't run the noopt tests, so that's the only way to test

Reply:
Oh, I didn't know that.
I created `config.toml` with `optimize-tests = false` and ran `x.py test src/test/ui --stage=1 --bless --config config.toml`, the test seems passed.